webdriver/protocol.py

```diff
@@ -0,0 +1,130 @@
+import contextlib
+import httplib
+import json
+import pytest
+import types
+
+import webdriver
+
+
+class HTTPRequest(object):
+    def __init__(self, host, port):
+        self.host = host
+        self.port = port
+
+    def head(self, path):
+        return self._request("HEAD", path)
+
+    def get(self, path):
+        return self._request("GET", path)
+
+    @contextlib.contextmanager
+    def _request(self, method, path):
+        conn = httplib.HTTPConnection(self.host, self.port)
+        try:
+            conn.request(method, path)
+            yield conn.getresponse()
+        finally:
+            conn.close()
+
+
+@pytest.fixture
+def http(request, session):
+    return HTTPRequest(session.transport.host, session.transport.port)
+
+
+@pytest.fixture
+def new_session(session):
+    session.end()
+    assert session.session_id is None
+    return session
+
+
+def test_provides_http(http):
+    # 4.0
+    with http.head("/") as resp:
+        assert resp.status == 404
+
+
+def test_send_an_error(http):
+    # 4.3 send an error, steps 1-5
+    with http.get("/cream") as resp:
```

Comment:
This assumes that the webdriver remote end is listening at the root of the server. From section 6.4:

> A remote end has an associated URL prefix, which is used as a prefix on all WebDriver-defined URLs on that remote end. This must either be undefined or an path-absolute URL.

We should be prefixing URLs with the URL prefix.